src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java

```diff
@@ -78,109 +138,187 @@ void equality() {
+        for (int i = 0; i < encodedReceipts.length; i++) {
             encodedReceipts[i] = Hex.toHexString(receiptMerkleProof.get(i).toMessage());
         }
 
-        String[] receipts = ((PowHSMSignerMessage) message).getReceiptMerkleProof();
+        String[] receipts = message.getReceiptMerkleProof();
 
         assertArrayEquals(encodedReceipts, receipts);
         assertNotSame(encodedReceipts, receipts);
     }
 
     @Test
     void getSigHash() {
-        String rawTx = "020000000001017001d967a340069c0b169fcbeb9cb6e0d78a27c94a41acbce762abc695aefab10000000017160014c" +
-                "fa63de9979e2a8005e6cb516b86202860ff3971ffffffff0200c2eb0b0000000017a914291a7ddc558810708149a731f39cd3c3" +
-                "a8782cfd870896e1110000000017a91425a2e67511a0207c4387ce8d3eeef498a4782e64870247304402207e0615f440bbc5035" +
-                "1fb5d8839b3fae6c74f652c9ffc9291008f4ea39f9565980220354c734511a0560367b300eecb1a7472317a995462622e06ee91" +
-                "cbe0517c17e1012102e87cd90f3cb0d64eeba797fbb8f8ceaadc09e0128afbaefb0ee9535875ea395400000000";
-        BtcTransaction btcTx = new BtcTransaction(RegTestParams.get(), Hex.decode(rawTx));
+        BtcTransaction btcTx = new BtcTransaction(RegTestParams.get(), Hex.decode(simpleRawBtcTx));
         int inputIndex = 0;
         TransactionReceipt txReceipt = new TransactionReceipt();
         List<Trie> receiptMerkleProof = new ArrayList<>();
         receiptMerkleProof.add(new Trie());
-        Sha256Hash sigHash = Sha256Hash.wrap("0000000000000000000000000000000000000000000000000000000000000001");
-        SignerMessage message = new PowHSMSignerMessage(btcTx, inputIndex, txReceipt, receiptMerkleProof, sigHash);
+        Sha256Hash sigHash = Sha256Hash.wrap(
+            "0000000000000000000000000000000000000000000000000000000000000001");
+        PowHSMSignerMessage message = new PowHSMSignerMessage(btcTx, inputIndex, txReceipt,
+            receiptMerkleProof, sigHash);
+
+        assertEquals(sigHash, message.getSigHash());
+    }
+
+    @ParameterizedTest
+    @MethodSource("getMessageToSignLegacyArgProvider")
+    void getMessageToSign_whenSighashLegacyMode_ok(int version, JsonNode expectedMessageToSend) {
+        // arrange
+        TransactionReceipt txReceipt = new TransactionReceipt();
+        List<Trie> receiptMerkleProof = new ArrayList<>();
+        receiptMerkleProof.add(new Trie());
+        Sha256Hash sigHash = BitcoinTestUtils.createHash(1);
+        PowHSMSignerMessage powHSMSignerMessage = new PowHSMSignerMessage(pegoutBtcTx,
+            FIRST_OUTPUT_INDEX, txReceipt, receiptMerkleProof, sigHash);
 
-        assertEquals(sigHash, ((PowHSMSignerMessage) message).getSigHash());
+        // act
+        JsonNode actualMessageToSign = powHSMSignerMessage.getMessageToSign(version);
+
+        // assertions
+        Assertions.assertEquals(expectedMessageToSend, actualMessageToSign);
+    }
+
+    @Test
+    void getMessageToSign_whenSighashSegwitMode_ok() {
```

Comment:
Maybe we should have a test where we sign a different input than just the first one?

Or a more complete test where we create the message to sign for more than one input

Reply:
Actually, this test is signing a pegout with three inputs.